src/setup.ts

```diff
@@ -1,6 +1,51 @@
-import internalSetup from './setup/internal'
+import { Endpoints, setup as internalSetup } from './setup/internal'
 
 
-export function ipfs(s: { [key: string]: unknown }): void {
-  internalSetup.ipfs = s
+type SomeEndpoints = {
```

Comment:
typescript has a nifty type modifier `Partial` where you can do `Partial<Endpoints>` and not have to define a new type:
https://www.typescriptlang.org/docs/handbook/utility-types.html#partialt

Reply:
Oh nice, will change 👍